codalab-service.py

```diff
@@ -0,0 +1,387 @@
+#! /usr/bin/python2.7
+
+import argparse
+import errno
+import os
+import subprocess
+import test_cli
+
+
+def main():
+    args = CodalabArgs.get_args()
+    service_manager = CodalabServiceManager(args)
+    service_manager.execute()
+
+
+class CodalabArgs(argparse.Namespace):
+    DEFAULT_ARGS = {
+        'version': 'latest',
+        'dev': False,
+        'push': False,
+        'docker_user': None,
+        'docker_pwd': None,
+        'build_locally': False,
+        'image': 'all',
+        'test_build': False,
+        'user_compose_file': None,
+        'start_worker': False,
+        'initial_config': False,
+        'root_dir': None,
+        'instance_name': 'codalab',
+        'mysql_root_pwd': 'mysql_root_pwd',
+        'mysql_user': 'codalab',
+        'mysql_pwd': 'mysql_pwd',
+        'root_user': 'codalab',
+        'root_pwd': 'testpassword',
+        'uid': None,
+        'service_home': None,
+        'mysql_mount': None,
+        'worker_dir': None,
+        'bundle_stores': [],
+        'worker_docker_network_name': 'codalab_worker_network',
+        'http_port': '80',
+        'rest_port': None,
+        'frontend_port': None,
+        'mysql_port': None,
+        'use_ssl': False,
+        'ssl_cert_file': None,
+        'ssl_key_file': None,
+        'follow': False,
+        'tail': None,
+        'tests': ['default'],
+    }
+
+    ARG_TO_ENV_VAR = {
```

Comment:
Could we combine this and the block above, so we have one place where all the options are defined?

Reply:
Actually better yet, can we just merge that with the options parsing block below, so that everything is in one place?  Especially having the documentation right would be really helpful.

Reply:
See my comment below about env vars. If we don't allow env var-based configuration this could be a simple argparse thing.

Reply:
I think as long as we can read it from a file, I don't think we need to support var-based config.  One other thing I thought of: one thing that would be helpful is to have it print out the environment variables that it's setting so if people want to do manual override, it's easy for them.  Similarly, printing out the docker commands that are run (maybe in a dry-run way) would be useful, so we have some flexibility.

Reply:
The problem is, the end commands end up having two parts to them (a set of env vars and the actual command), so even if we printed the commands being executed, just running those wouldn't be enough. We could print the env vars AND the commands (in the copyable `VAR=val VAR2=val2 cmd --args` format) but these would be huge strings given the complexity of env vars and commands being run.
Maybe we can add a verbosity flag to enable printing these?